src/compile/model.ts

```diff
@@ -361,6 +361,20 @@ export abstract class Model {
   public getComponent(type: 'scales' | 'selection', name: string): any {
     return this.component[type][name] || this.parent.getComponent(type, name);
   }
+
+  public getLookups(predicate?: (lookup: LookupTransform) => boolean): LookupTransform[] {
+    const lookups: LookupTransform[] = this.transforms.filter((t: Transform): t is LookupTransform => isLookup(t) && (!predicate || predicate(t)));
+    return lookups;
+  }
+}
+
+// above works if we add this, though it should be already added from typescript!!!
+// See this PR https://github.com/Microsoft/TypeScript/issues/7657 which was merged in ~ TS 2.1
+// TODO: remove this before merge. If we can't figure out TS, just have getLookups return Transform[]
+declare global {
```

Comment:
We should be using ts 2.3. I don't really understand the problem or have a solution.